Do not checkout the branch passed to `CommitFileViaConsole`

weierophinney opened the pull request:
We need to assume we're on the correct branch; otherwise, a checkout operation will lead to an error, aborting the operation.

Instead, I've added functionality to verify that we are on the branch provided; if not, an exception is raised.

The changes also required annotating that the repository directory, filename, and commit message arguments could not be empty, which required changes elsewhere to ensure this is true before calling on `CommitFile`.

weierophinney committed:
fix: do not checkout the branch passed to `CommitFileViaConsole`

We need to assume we're on the correct branch; otherwise, a checkout
operation will lead to an error, aborting the operation.

Instead, I've added functionality to verify that we are on the branch
provided; if not, an exception is raised.

The changes also required annotating that the repository directory,
filename, and commit message arguments could not be empty, which
required changes elsewhere to ensure this is true before calling on
`CommitFile`.

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
Adds CHANGELOG entry for #41

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

Ocramius commented on src/Changelog/BumpAndCommitChangelogVersionViaKeepAChangelog.php:
```diff
@@ -66,11 +66,14 @@ public function __invoke(
         $bumper->updateChangelog($newVersion);
 
+        $message = sprintf(self::COMMIT_TEMPLATE, $newVersion, self::CHANGELOG_FILE, $newVersion);
+        Assert::stringNotEmpty($message);
```
I think this is just `Assert::notEmpty()`, since `sprintf(): string` already enforces it to be a `string` 

Doesn't psalm complain about this one?

weierophinney replied:
It didn't - it complained that it couldn't determine it was non-empty, which is why I added the assertion. I'll update to simply `notEmpty()` and see what happens.

Ocramius commented on src/Changelog/CommitReleaseChangelogViaKeepAChangelog.php:
```diff
@@ -69,11 +70,14 @@ public function __invoke(
         }
 
+        $message = sprintf(self::COMMIT_TEMPLATE, $versionString, self::CHANGELOG_FILE);
+        Assert::stringNotEmpty($message);
```
Same comment as above: need to `Assert::notEmpty()` only

Ocramius commented on src/Git/CommitFileViaConsole.php:
```diff
@@ -26,4 +29,30 @@ public function __invoke(
+     * @param non-empty-string $repositoryDirectory
+     * @param non-empty-string $filename
+     */
+    private function assertBranch(
```
`assertBranch()` naming is a bit generic: would suggest naming this something like `assertWeAreOnBranch...` or similar

Ocramius commented on src/Git/CommitFileViaConsole.php:
```diff
@@ -26,4 +29,30 @@ public function __invoke(
+        string $filename
+    ): void {
+        $process = new Process(['git', 'branch', '--show-current'], $repositoryDirectory);
+        $process->run();
```
You want to call `$process->mustRun()` here: that already throws meaningful exceptions with STDOUT/STDERR as part of them (no need for the assertion either)

weierophinney committed:
qa: use notEmpty instead of stringNotEmpty when we already know we have a string

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney committed:
qa: better method name

Renamed `assertBranch` to `assertWeAreOnBranch`. Additionally, instead
of:

- Process:run
- Process:isSuccessful
- throw

use just `Process::mustRun()`

Signed-off-by: Matthew Weier O'Phinney <matthew@weierophinney.net>

weierophinney commented:
Feedback incorporated!

Ocramius reviewed (approved):
LGTM :+1:
